src/bun.js/bindings/OnigurumaRegExp.cpp

```diff
@@ -46,6 +54,35 @@ static  WTF::String to16Bit(JSValue jsValue, JSC::JSGlobalObject *globalObject,
     return to16Bit(jsString, globalObject);
 }
 
+static WTF::String extendMultibyteHexCharacters(const WTF::String &string) {
+    WTF::StringBuilder sb;
+    bool escape = false;
+    bool hex = false;
+    int hDigitCounter = 0;
+    for (int i = 0; i < string.length(); i++) {
+        sb.append(string.characterAt(i));
+        if (string.characterAt(i) == '\\') {
+            escape = true;
+        } else if (string.characterAt(i) == 'x') {
+            hex = true;
+        } else if (string.characterAt(i) == '{' && hex && escape) {
+            hex = false;
+            escape = false;
+            hDigitCounter = 0;
+        } else if (isxdigit(string.characterAt(i)) && escape && hex) {
+            hDigitCounter += 1;
+            if (hDigitCounter == 2) {
+                hDigitCounter = 0;
+                escape = false;
+                hex = false;
+                sb.append("\\x00"_s);
+            }
+        }
```

Comment:
What if you give it `"\\nx00"`?

Instead of setting `hex` and `escape` above the loop, consider incrementing inside the place that's being checked